axelor-sale/src/main/resources/views/SaleOrderLine.xml

```diff
@@ -1046,4 +1050,11 @@
     <attribute for="product" name="focus" expr="true"/>
   </action-attrs>
 
+  <action-record
+    name="action-sale-order-line-record-set-default-invoicing-mode-select"
+    model="com.axelor.apps.sale.db.SaleOrderLine">
+    <field name="invoicingModeSelect" expr="eval:1" if="toInvoice"/>
```

Comment:
You can get the repository value as such instead of using 1 :
__ repo __(SaleOrderLine).INVOICING_MODE_DIRECTLY (remove spaces around the word 'repo')